common/transport/amqp/src/amqp_link_interface.ts

```diff
@@ -1,5 +1,7 @@
+import { AmqpError } from 'rhea';
+
 export interface AmqpLink {
   attach: (callback: (err?: Error) => void) => void;
-  detach: (callback: (err?: Error) => void) => void;
+  detach: (callback: (err?: Error) => void, err?: Error | AmqpError) => void;
```

Comment:
>callback: (err?: Error) => void, err?: Error | AmqpError [](start = 11, length = 56)

argument order should be reversed. I'm ok with breaking the API, a lot less ok with breaking the callback pattern